examples/reference/custom_components/CompositeWidget.md

```diff
@@ -0,0 +1,129 @@
+# `CompositeWidget`
+
+`CompositeWidget` simplifies the creation of custom Panel widgets using only Python and Panel components.
+
+```{pyodide}
+import param
+import panel as pn
+from panel.widgets import CompositeWidget
+
+pn.extension()
+
+class ButtonGroup(CompositeWidget):
```

Comment:
It would have been really nice of the `Viewer` and the `CompositeWidget` had similar names and worked in similar ways.

I believe it would have made things simpler if 

- they had had the names `CustomComponent` and `CustomWidget` respectively.
- they had worked in a similar way either via a `_composite` parameter or a `__panel__` method.
- that the `Viewer` was a `Layoutable` and also had its `Layoutable` parameters synced with the component shown. Right now you have to write a lot of boilerplate to get that working.

I don't know if now is the time to fix this friction?

Reply:
Yes, all solid points. `CompositeWidget` was really developed for internal usage and never advertised while `Viewer` was actually advertised as an official mechanism for extending Panel.

Reply:
Would it be possible to refactor Viewer internals to inherit from CompositeWidget?

Reply:
The problem is that it will then be a breaking change because the `CompositeWidget` and `Viewer` have different APIs.

But I agree we should simplify the story and chose one of the APIs for both widgets and non-widgets.

Reply:
Hmm. I confused about which class a `PyComponent` class should inherit from? Panel mixes in a lot in the `Widget` and `Pane` base classes.